### PR TITLE
Sectioned create a question page into parts with a follow up separation of the answers and references. This for UI presentation purposes but also for breakout into constituent components.

### PR DESCRIPTION
**Pull request by BrianHCombes**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -40,6 +42,8 @@ import { GenerateExamPageComponent } from './generate-exam-page/generate-exam-pa
 import { SearchForAnExamPageComponent } from './search-for-an-exam-page/search-for-an-exam-page.component';
 import { CreateExamPageComponent } from './create-exam-page/create-exam-page.component';
 import { CreateQuestionPageComponent } from './create-question-page/create-question-page.component';
+import { CreateQuestionPageComponentPart1 } from './create-question-page/create-question-page-part1/create-question-page-part1.component';
```
Comment (haxwell) [inline src/app/app.module.ts]: A better name would be `CreateQuestionPageComponent.QuestionSection`... something more descriptive, `..Part1`. Nobody who comes behind you working on this code will know what `..Part1` means.

File: src/app/_services/api.service.ts [hunk 2]
```diff
@@ -10,58 +10,71 @@ const baseUrl = environment.apiUrl;
...
-      // The response body may contain clues as to what went wrong,
-      console.log("In handleError: error= " + JSON.stringify(error));
-      console.error(
-        `Backend returned code ${error.status}, ` +
-        `body was: ${error.error}`);
+      var username = "johnathan";
+      var password = "password";
+      // alert(baseUrl + "" + url);
+      // return this.http.post(baseUrl + url, data, this.getHttpHeadersWithBase64Authorization(username, password))
+      return this.http.post('http://localhost:8080/api/question', data, this.getHttpHeadersWithBase64Authorization(username, password));
```
Comment (haxwell) [inline src/app/_services/api.service.ts]: Read this line carefully. Remember this is going to Production, live on the internet. What is wrong with this line?

File: src/app/_services/api.service.ts [hunk 3]
```diff
@@ -10,58 +10,71 @@ const baseUrl = environment.apiUrl;
...
+        this.getHttpHeadersWithBase64Authorization(username, password))
+        .pipe(
+          catchError(this.handleError)
+        );
+    }
 
-  post(url, data) {
-  	return this.http.post(baseUrl + url, data, this.getHttpHeaders())
-  }
+/*
```
Comment (haxwell) [inline src/app/_services/api.service.ts]: You usually should not commit commented out code. If you just want to remember what it was like before you changed it, that is what source control (git) is for.

File: src/app/_services/api.service.ts [hunk 4]
```diff
@@ -10,58 +10,71 @@ const baseUrl = environment.apiUrl;
...
 
-  getHttpHeaders() {
-  	let rtn = {
-  		headers: new HttpHeaders({
-  			'Content-Type': 'application/json'
-  		})
-  	}
+    getHttpHeaders() {
+          let rtn = {
+                  headers: new HttpHeaders({
```
Comment (haxwell) [inline src/app/_services/api.service.ts]: These changes with the bars on each line, indicate whitespace changes. Meaning you added a tab, or spaces, or you deleted them; minor changes like that. Likely, you ran some kind of formatting on this file. It is fine if you do that, but you should not also make code changes to the file at the same time. Do the formatting in one commit, and the code changes in another.  Do one thing.

Revert your changes to this file, add the valid change you made to the `post()` method, and commit that single change as part of your PR.

File: src/app/create-question-page/create-question-page.component.ts [hunk 5]
```diff
@@ -1,15 +1,143 @@
...
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatSnackBar } from '@angular/material';
+
+import { ApiService } from './../_services/api.service';
 
 @Component({
-  selector: 'app-create-question-page',
-  templateUrl: './create-question-page.component.html',
-  styleUrls: ['./create-question-page.component.css']
+    selector: 'app-create-question-page',
```
Comment (haxwell) [inline src/app/create-question-page/create-question-page.component.ts]: Again, formatting change.. should be in its own commit.

**Comment by haxwell** [comment]
Its coming along, Brian! There is a lot of good work in here, I'd love to get it committed team-wide. Tackle those comments I left, and we should be good!

**Comment by BrianHCombes** [comment]
Johnathan,
    Thanks for taking a look at #48 and pointing out the needed fixes and adjustments. I'll get it all cleaned up and resubmitted for another look.

    Yes, I just looked and I see all the comments you left with the PR at the repo. Was likely not paying close enough attention previously and blundered through, but now seem to have a mental fix on it. 
    Thanks!

    Brian
    On Wednesday, January 23, 2019, 9:22:44 AM MST, Johnathan James <notifications@github.com> wrote:  
 
 
Its coming along, Brian! There is a lot of good work in here, I'd love to get it committed team-wide. Tackle those comments I left, and we should be good!

File: src/app/app.module.ts (hunk 1 above)
Comment (BrianHCombes) [reply src/app/app.module.ts]: Have changed the component name to follow existing nomenclature on upcoming re-submission.

File: src/app/_services/api.service.ts (hunk 2 above)
Comment (BrianHCombes) [reply src/app/_services/api.service.ts]: Have cleaned this up with proper sourcing of baseUrl and endpoint url

File: src/app/_services/api.service.ts (hunk 3 above)
Comment (BrianHCombes) [reply src/app/_services/api.service.ts]: Removed.

File: src/app/_services/api.service.ts (hunk 4 above)
Comment (BrianHCombes) [reply src/app/_services/api.service.ts]: Understood and hopefully can find and undo these (spaces and adjustments)  and then re-apply them within the code procedures you recommend. A learnt moment. Thanks!

File: src/app/create-question-page/create-question-page.component.ts (hunk 5 above)
Comment (BrianHCombes) [reply src/app/create-question-page/create-question-page.component.ts]: Understood and hopefully can find and undo these (spaces and adjustments)  and then re-apply them within the code procedures you recommend. A learnt moment. Thanks!
